book/chapters/config.md

```diff
@@ -2,15 +2,49 @@
 
 ## Reading a configure file
 
-You can tell your app to allow configure files with `set_config("--config")`. There are arguments: the first is the option name. If empty, it will clear the config flag. The second item is the default file name. If that is specified, the config will try to read that file. The third item is the help string, with a reasonable default, and the final argument is a boolean (default: false) that indicates that the configuration file is required and an error will be thrown if the file
-is not found and this is set to true.
+You can tell your app to allow configure files with `set_config("--config")`. There are arguments: the first is the option name. If empty, it will clear the config flag. The second item is the default file name. If that is specified, the config will try to read that file. The third item is the help string, with a reasonable default, and the final argument is a boolean (default: false) that indicates that the configuration file is required and an error will be thrown if the file is not found and this is set to true.
+
+### Extra fields
+Sometimes configuration files are used for multiple purposes so CLI11 allows options on how to deal with extra fields
+
+```cpp
+app.allow_config_extras(true);
+```
+will allow capture the extras in the extras field of the app. (NOTE:  This also sets the allow_extras in the app to true)
+
+```cpp
+app.allow_config_extras(false);
+```
+will generate an error if there are any extra fields
+
+for slightly finer control there is a scoped enumeration of the modes
+or
+```cpp
+app.allow_config_extras(CLI::config_extras_mode::ignore);
+```
+will completely ignore extras and is the default.
+
+```cpp
+app.allow_config_extras(CLI::config_extras_mode::capture);
```

Comment:
Is this equivalent to true?

Reply:
It is not exactly equivalent.  For backwards compatibility setting to true also sets the allow_extras flag,  So it adds some coupling between the flags which may or may not be desired.  Using the enumeration doesn't have any coupling.

Reply:
Maybe adding this note would be useful, then. At least somewhere (in doxygen comment, for example).